p2p/p2p.go

```diff
@@ -156,13 +161,32 @@ func (s *Service) Stop() error {
 		return err
 	}
 
-	if s.msgChan != nil {
-		close(s.msgChan)
+	if s.msgSendChan != nil {
+		close(s.msgSendChan)
 	}
 
 	return nil
 }
 
+// Start polling the msgRecChan channel for any blocks
+func (s *Service) MsgRecPoll() (err error) {
+	for {
+		// Receives block from babe
+		block := <-s.msgRecChan
+		msg := s.hostAddr.String() + " received block"
+		log.Info(msg, "block", block)
+
+		// Calls broadcast
+		for _, peers := range s.host.Network().Peers() {
```

Comment:
You should be using `Broadcast` here. I see the parameter type doesn't match up, maybe we change the channel to be of type `Message`?

Reply:
we could also change Broadcast/Send to take []byte

Reply:
Broadcast needs to know what msg type it is (right now)

Reply:
that can also be done by reading the first byte of the message to be sent and comparing it to the diff message types